### PR TITLE
JsBox should be able to load URL using Basic Auth.

### PR DESCRIPTION
**Pull request by smn**
I'm not sure if GitHub allows for basic auth access to stuff to begin with but that's a separate issue.

### CODE REVIEW

File: go/apps/jsbox/tests/test_views.py [hunk 1]
```diff
@@ -60,3 +62,31 @@ def test_edit_conversation(self):
...
+
+    @patch('requests.get')
+    def test_basic_auth_cross_domain_xhr(self, mocked_get):
+        mocked_get.return_value = Mock(text='foo', status_code=200)
+        response = self.client.post(
+            reverse('jsbox:cross_domain_xhr'),
+            {'url': 'http://username:password@domain.com'})
+        [call] = mocked_get.call_args_list
+        args, kwargs = call
+        self.assertEqual(args, ('http://username:password@domain.com',))
```
Comment (hodgestar) [inline go/apps/jsbox/tests/test_views.py]: Seems a bit odd that the username and password are still in the URL? Don't think it'll directly harm anyone but maybe worth a comment and a little extra thought?

**Comment by hodgestar** [comment]
Are people okay with their passwords being stored as plain-text in config? Probably not much we can do to avoid it for this kind of delegated login.

**Comment by smn** [comment]
I don't see any way around this. HTTP Basic Auth requires us to store passwords in plain text, we can add a bit of text that tells people this and make a strong suggestion to use a system that uses revokable-tokens for passwords but that's about the best we can do.

**Comment by hodgestar** [comment]
Any reason to not land this and just require that people use GitHub's Oauth
2 support?

**Comment by smn** [comment]
People should be free to host their JS anywhere they like, not just GitHub.

File: go/apps/jsbox/tests/test_views.py (hunk 1 above)
Comment (hodgestar) [inline go/apps/jsbox/tests/test_views.py]: Just poking this comment again.

**Comment by hodgestar** [comment]
I think I'm +1 but I'd like GitHub users to use the Oauth2 `access_token` parameter from http://developer.github.com/v3/#authentication.

File: go/apps/jsbox/tests/test_views.py (hunk 1 above)
Comment (smn) [inline go/apps/jsbox/tests/test_views.py]: Thanks! Fixed.

**Comment by smn** [comment]
Ready for re-review @hodgestar

**Comment by hodgestar** [comment]
:+1:

**Pull request merged**